cmd/contour/serve.go

```diff
@@ -133,10 +133,13 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	contourInformers := contourinformers.NewSharedInformerFactory(clients.contour, 0)
 
 	// Create a set of SharedInformerFactories for each root-ingressroute namespace (if defined)
-	var namespacedInformers []coreinformers.SharedInformerFactory
+	namespacedInformers := map[string]coreinformers.SharedInformerFactory{}
+
 	for _, namespace := range ctx.ingressRouteRootNamespaces() {
-		inf := coreinformers.NewSharedInformerFactoryWithOptions(clients.core, 0, coreinformers.WithNamespace(namespace))
-		namespacedInformers = append(namespacedInformers, inf)
+		if _, ok := namespacedInformers[namespace]; !ok {
```

Comment:
I like this as a map, it also protects from users entering the same namespace multiple times! =)